### PR TITLE
[4.0] Compute the controller name correctly

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Comment #https://github.com/joomla/joomla-cms/pull/17935#issuecomment-331387465.

### Summary of Changes
Changing the status in the list view fails.

### Testing Instructions
Publish an article in the list view.

### Expected result
It works and returns to the list view.

### Actual result
It throws an error.

### CODE REVIEW

File: libraries/src/MVC/Controller/AdminController.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public function __construct($config = array(), MVCFactoryInterface $factory = nu
 		{
 			$reflect = new \ReflectionClass($this);
 
-			$r = array(0 => '', 1 => '', 2 => $reflect->getShortName());
+			$r = array(0 => '', 1 => '', 2 => str_replace('Controller', '', $reflect->getShortName()));
```
Comment (wilsonge) [inline libraries/src/MVC/Controller/AdminController.php]: You can't do this until you're sure it's a namespaced class?

**Comment by wilsonge** [comment]
@infograf768 can you just confirm this is ok?

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 303c54a2bd9f034efc80647b56756a08cd8a7266<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18063">issues.joomla.org/tracker/joomla-cms/18063</a>.</sub>

**Comment by infograf768** [comment]
Also please re-add the strings


COM_CONTACT_N_ITEMS_UNPUBLISHED="%d contacts unpublished."
COM_CONTACT_N_ITEMS_UNPUBLISHED_1="%d contact unpublished."
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18063">issues.joomla.org/tracker/joomla-cms/18063</a>.</sub>

**Comment by laoneo** [comment]
Are they related to this pr?

**Comment by infograf768** [comment]
not specifically, found they were missing when testing

**Comment by wilsonge** [comment]
Let's do that in another PR then :)

**Pull request merged**